lib/src/database/database_api.dart

```diff
@@ -357,6 +355,18 @@ abstract class DatabaseApi {
 
   Future<DiscoveryInformation?> getWellKnown();
 
+  Future<void> storeDeviceId(String deviceId);
+
+  Future<String?> getDeviceId();
+
+  Future<void> storeOidcAuthMetadata(Map<String, Object?>? authMetadata);
+
+  Future<Map<String, Object?>?> getOidcAuthMetadata();
+
+  Future<void> storeOidcDynamicClientId(String? oidcClientId);
+
+  Future<String?> getOidcDynamicClientId();
```

Comment:
I don't think we need this. Just use `getClient()` and pick what you need from it. To have it type safe we could later migrate it to returning a Dart Record